### PR TITLE
refacotry testcase for project member manage

### PR DESCRIPTION
**Pull request by ninjadq**
Refactory testcase for project member management
  * Add some keyword for simplifying the case.
  * Replace Test Case-Manage Project Member on `BAT.robot`
  * Also put this case on file `3-03-DB-manage-project-members` in Group3-RBAC

### CODE REVIEW

**Comment by reasonerjt** [comment]
Please squash the commits.

**Comment by ninjadq** [comment]
@reasonerjt squashed
and replaced all tabs to space

File: tests/resources/Harbor-Pages/Project-Members.robot [hunk 1]
```diff
@@ -21,69 +21,254 @@ ${HARBOR_VERSION}  v1.1.1
...
-	Sleep  2
-	Wait Until Page Contains  ${role}
-	
-	
-	
\ No newline at end of file
+    Sleep  2
+    Wait Until Page Contains  ${role}
+    
+User Can Change Role
```
Comment (wy65701436) [inline tests/resources/Harbor-Pages/Project-Members.robot]: can we just separate the xpath data from function. It would be better to add them into Project_Member_Element.robot

File: tests/resources/Harbor-Pages/Project-Members.robot [hunk 2]
```diff
@@ -21,69 +21,254 @@ ${HARBOR_VERSION}  v1.1.1
...
+     Page Should Contain Element  xpath=//project-detail//clr-dg-row-master[contains(.,'${username}')]//clr-dg-action-overflow
+User Can Not Change Role
+     [arguments]  ${username}
+     Page Should Contain Element  xpath=//project-detail//clr-dg-row-master[contains(.,'${username}')]//clr-dg-action-overflow[@hidden=""]
+
+Non-admin View Member Account
+    [arguments]  ${times}
+    Xpath Should Match X Times  //project-detail//clr-dg-action-overflow[@hidden=""]  ${times}
+
+User Can Not Add Member
```
Comment (wy65701436) [inline tests/resources/Harbor-Pages/Project-Members.robot]: what's does this func mean?

File: tests/resources/Harbor-Pages/Project-Members.robot [hunk 3]
```diff
@@ -21,69 +21,254 @@ ${HARBOR_VERSION}  v1.1.1
...
+    Sign In Harbor  ${HARBOR_URL}  usera${d}  Test1@34
+    #create project
+    Create An New Project  project${d}
+    #verify can not change role
+    Go Into Project  project${d}
+    Switch To Member
+    User Can Not Change Role  usera${d}
+    Logout Harbor
+    #login console as usera and push
+    ${rc}=  Run And Return Rc  docker pull hello-world
```
Comment (wy65701436) [inline tests/resources/Harbor-Pages/Project-Members.robot]: we do have an func() named PushImage.

File: tests/resources/Harbor-Pages/Project-Members.robot [hunk 4]
```diff
@@ -21,69 +21,254 @@ ${HARBOR_VERSION}  v1.1.1
...
+
+User A Create New Project And Push Image To It
+    [Arguments]  ${ip}  ${d}
+    Sign In Harbor  ${HARBOR_URL}  usera${d}  Test1@34
+    #create project
+    Create An New Project  project${d}
+    #verify can not change role
+    Go Into Project  project${d}
+    Switch To Member
+    User Can Not Change Role  usera${d}
```
Comment (wy65701436) [inline tests/resources/Harbor-Pages/Project-Members.robot]: why add this key words here? Does this func is just to push image to the newly created project?

File: tests/resources/Harbor-Pages/Project-Members.robot [hunk 5]
```diff
@@ -21,69 +21,254 @@ ${HARBOR_VERSION}  v1.1.1
...
+    [Arguments]  ${d}
+    Create An New User  ${HARBOR_URL}  username=usera${d}  email=usera${d}@vmware.com  realname=usera${d}  newPassword=Test1@34  comment=harbor
+    Logout Harbor
+    Create An New User  ${HARBOR_URL}  username=userb${d}  email=userb${d}@vmware.com  realname=userb${d}  newPassword=Test1@34  comment=harbor
+    Logout Harbor
+    Create An New User  ${HARBOR_URL}  username=userc${d}  email=userc${d}@vmware.com  realname=userc${d}  newPassword=Test1@34  comment=harbor
+    Logout Harbor
+
+User A Create New Project And Push Image To It
+    [Arguments]  ${ip}  ${d}
```
Comment (wy65701436) [inline tests/resources/Harbor-Pages/Project-Members.robot]: add project name and image name into the parameter list.

File: tests/resources/Harbor-Pages/Project-Members.robot [hunk 6]
```diff
@@ -21,69 +21,254 @@ ${HARBOR_VERSION}  v1.1.1
...
+    ${rc}=  Run And Return Rc  docker tag hello-world ${ip}/project${d}/bbbbb
+    ${rc}=  Run And Return Rc  docker push ${ip}/project${d}/bbbbb
+    Should Not Be Equal As Integers  ${rc}  0
+    #login ui as b
+    Sign In Harbor  ${HARBOR_URL}  userb${d}  Test1@34
+    Project Should Not Display  project${d}
+    Logout Harbor
+ 
+User A Add User B As Guest Role On Project
+    [Arguments]  ${d}
```
Comment (wy65701436) [inline tests/resources/Harbor-Pages/Project-Members.robot]: userA and userB and project should be the parameter?

File: tests/resources/Harbor-Pages/Project-Members.robot [hunk 7]
```diff
@@ -21,69 +21,254 @@ ${HARBOR_VERSION}  v1.1.1
...
+    Logout Harbor
+    #login console as usera and push
+    ${rc}=  Run And Return Rc  docker pull hello-world
+    ${rc}  ${output}=  Run And Return Rc And Output  docker login -u usera${d} -p Test1@34 ${ip}
+    ${rc}=  Run And Return Rc  docker tag hello-world ${ip}/project${d}/hello-world
+    ${rc}=  Run And Return Rc  docker push ${ip}/project${d}/hello-world
+    Should Be Equal As Integers  ${rc}  0
+    ${rc}=  Run And Return Rc  docker logout ${ip}
+
+User B Not Project Member Cannot See And Operate The Project
```
Comment (wy65701436) [inline tests/resources/Harbor-Pages/Project-Members.robot]: reword it to give a meaningful name?

File: tests/resources/Harbor-Pages/Project-Members.robot [hunk 8]
```diff
@@ -21,69 +21,254 @@ ${HARBOR_VERSION}  v1.1.1
...
+    Logout Harbor
+
+User A Change Role of User B To Developer
+    [Arguments]  ${d}
+    Sign In Harbor  ${HARBOR_URL}  usera${d}  Test1@34
+    #change userb to developer
+    Change Project Member Role  project${d}  userb${d}  Developer
+    Logout Harbor
+
+User B As Developer Can Manage Image But Cannot Manage Member
```
Comment (wy65701436) [inline tests/resources/Harbor-Pages/Project-Members.robot]: I do believe all the keywords are all reusable instead of a description of a piece of code.
Let's discuss.

File: tests/resources/Harbor-Pages/Project-Members.robot [hunk 9]
```diff
@@ -21,69 +21,179 @@ ${HARBOR_VERSION}  v1.1.1
...
+
+Delete project member
+    [arguments]  ${member}
+    Click Element  xpath=//project-detail//clr-dg-row-master[contains(.,'${member}')]//clr-dg-action-overflow
+    Click Element  xpath=//project-detail//clr-dg-cell//clr-dg-action-overflow//button[contains(.,"Delete")]   
+    Sleep  1
+    Click Element  xpath=//confiramtion-dialog//button[2]
+    Sleep  1
+
+Create Three Different Users
```
Comment (wy65701436) [inline tests/resources/Harbor-Pages/Project-Members.robot]: Nowhere to call this func?

File: tests/resources/Harbor-Pages/Project-Members.robot [hunk 10]
```diff
@@ -21,69 +21,179 @@ ${HARBOR_VERSION}  v1.1.1
...
+    User Can Not Change Role  ${user}
+    Push image  ${ip}  ${user}  ${pwd}  ${project}  hello-world
+    Logout Harbor
+
+User Should Not Be A Member Of Project
+    [Arguments]  ${user}  ${pwd}  ${project}
+    Sign In Harbor  ${HARBOR_URL}  ${user}  ${pwd}
+    Project Should Not Display  ${project}
+    Logout Harbor
+    ${rc}  ${output}=  Run And Return Rc And Output  docker login -u ${user} -p ${pwd} ${ip}
```
Comment (wy65701436) [inline tests/resources/Harbor-Pages/Project-Members.robot]: use the keyword Cannot Push Image?

File: tests/resources/Harbor-Pages/Project-Members.robot [hunk 11]
```diff
@@ -21,69 +21,179 @@ ${HARBOR_VERSION}  v1.1.1
...
+    Click Element  xpath=//project-detail//button//clr-icon
+    Sleep  1
+    Input Text  xpath=//add-member//input[@id="member_name"]  ${member}
+    #select guest
+    Mouse Down  xpath=//project-detail//form//input[@id='checkrads_guest']
+    Mouse Up  xpath=//project-detail//form//input[@id='checkrads_guest']
+    Click Button  xpath=//project-detail//add-member//button[2]
+    Sleep  1
+
+Delete project member
```
Comment (wy65701436) [inline tests/resources/Harbor-Pages/Project-Members.robot]: please use upper case in the keywords.
